### PR TITLE
add Solana chain ID workaround

### PR DESCRIPTION
**Pull request by glitch-txs**
# fix: 

- There's currently an issue on Solana integrations, we have an old and a new chain ID for Mainnet and Devnet. Some wallets and some dapps are using the old chain ID and vice versa.
This PR adds a patch to support both cases and throw a warning in the dapp is the wallet or the dapp are using the old chain.

Tested with WalletConnect Web Sample Wallet and Trust Wallet on Android. (Connection + Sign Message)

- When the session was deleted from the wallet side while the app was closed, on page reload it would show up as connected.

### CODE REVIEW

File: packages/solana/src/utils/chainPath/constants.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const SolanaChainIDs = {
+	Mainnet: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp',
```
Comment (enesozturk) [inline packages/solana/src/utils/chainPath/constants.ts:2]: What about the testnet? Don't we have to include it as well? Or maybe not any changes to it? @glitch-txs

Comment (glitch-txs) [reply packages/solana/src/utils/chainPath/constants.ts:2]: testnet ID was added last so no need

File: packages/solana/src/utils/chainPath/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+export function getDefaultChainFromSession(
+  session: SessionTypes.Struct,
+  selectedChain: ChainIDType
+) {
+  const chains = session.namespaces['solana']?.accounts.map(
+    (account: string) => `solana:${account.split(':')[1]}`
+  )
+
+  if (selectedChain === Chains.Mainnet) {
```
Comment (enesozturk) [inline packages/solana/src/utils/chainPath/index.ts:39]: @glitch-txs Shouldn't these be like the following:
```suggestion
  if (selectedChain === Chains.Mainnet_Deprecated) {
```

Comment (enesozturk) [reply packages/solana/src/utils/chainPath/index.ts:39]: Or okay this logic working another way

**Review by enesozturk — APPROVED** [review]

**Pull request merged**